services/frontend-service/src/ui/Warnings.test.tsx

```diff
@@ -0,0 +1,60 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2021 freiheit.com*/
+import React from 'react';
+import { getByLabelText, getByText, render } from '@testing-library/react';
+import { UndeployBtn } from './Warnings';
+
+describe('Undeploy Button', () => {
+    const getNode = (overrides?: {}): JSX.Element | any => {
+        // given
+        const defaultProps: any = {
+            children: null,
+            state: 'no state', //
+            applicationName: 'app1', //
+        };
+        return <UndeployBtn {...defaultProps} {...overrides} />;
+    };
+    const getWrapper = (overrides?: { state?: string }) => render(getNode(overrides));
+
+    it('renders the UndeployBtn component', () => {
```

Comment:
I think it's best to use table driven tests here since they all render the same container only with different states

https://github.com/freiheit-com/kuberpult/blob/76802b82f8c5ebf6065f524567cc1b8ca14f5f36/services/frontend-service/src/ui/Releases.test.tsx#L89-L133

Reply:
I tried to do that at first then I changed it because I need to use different selectors but the fact that I use the same validation function can help here and I can use it